modules/rate/rate.model.js

```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose');
+
+const rateSchema = new mongoose.Schema({
```

Comment:
I think it is not the best approach for the handling of ratings. From my perspective, it is better to have a field
`ratings: {   [userId]: [ratingValue]  }`. Per each product, we are going to have one rating map. Also, it is good to think, do we really need a separate collection for the ratings. Maybe it would be much easier to just include it inside the product model. Let's discuss it on the meeting

Reply:
Ooh, this is a really good idea. Granted, it'll have to be `ratings: { user: [userId], value: [ratingValue] }`, as a changeable value of `userId` cannot serve as a key in schema, but incorporating it in products this way should be way easier to handle. I'll check if there seem to be no hidden issues and modify accordingly or get back to you. Thank you for the suggestion